client.go

```diff
@@ -0,0 +1,66 @@
+package eywa
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+type Client struct {
+	endpoint   string
+	httpClient *http.Client
+	headers    map[string]string
+}
+
+// NewClient accepts a graphql endpoint and returns back a Client.
+// It uses the http.DefaultClient as the underlying http client by default.
+func NewClient(gqlEndpoint string) *Client {
+	return &Client{
+		endpoint:   gqlEndpoint,
+		httpClient: http.DefaultClient,
+	}
+}
+
+// WithHeaders adds the given headers to all the requests made by the Client.
+func (c *Client) WithHeaders(headers map[string]string) *Client {
+	c.headers = headers
+	return c
+}
+
+// WithHttpClient sets the underlying HTTP client that will be used for making graphql calls.
+func (c *Client) WithHttpClient(hc *http.Client) *Client {
```

Comment:
Maybe it is best to get this as an optional arg in NewClient. Because, this would allow anyone to modify the shared eywa client from any part of the program during runtime.